mitiq/zne/inference.py

```diff
@@ -34,16 +25,22 @@
 from mitiq.observable import Observable
 from mitiq.zne.scaling import fold_gates_at_random
 
-ExtrapolationResult = Union[
-    float,  # The zero-noise value.
-    Tuple[
-        float,  # The zero-noise value.
-        Optional[float],  # The (estimated) error on the zero-noise value.
-        List[float],  # Optimal parameters found during fitting.
-        Optional[np.ndarray],  # Covariance of fitting parameters.
-        Callable[[float], float],  # Function that was fit.
-    ],
-]
+# Extrapolation Result:
+# - float: The zero-noise value.
+# - float (optional): The (estimated) error on the zero-noise value.
+# - list[float]: Optimal parameters found during fitting.
+# - npt.NDArray[np.float64] (optional): Covariance of fitting parameters.
+# - Callable[[float], float]: Function that was fit.
+ExtrapolationResult: TypeAlias = (
+    float
+    | tuple[
+        float,
+        float | None,
+        list[float],
+        npt.NDArray[np.float64] | None,
+        Callable[[float], float],
+    ]
+)
```

Comment:
1. Can you format this as before? I think it's a little easier to read.
2. What does adding `TypeAlias` here do for us?